qiime/sdk/execution.py

```diff
@@ -0,0 +1,44 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016--, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file COPYING.txt, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import os
+import tempfile
+import subprocess
+
+
+class ExecutionContext:
+
+    def __call__(self):
+        # Should this be decorated with abc.abstract_method?
+        pass
+
+
+class LocalSubprocessExecution:
+
+    _python_executable = "python"
+
+    def __call__(self, workflow, input_artifact_filepaths,
+                 parameter_references, output_artifact_filepaths):
+        input_artifact_abs_filepaths = \
+            {k: os.path.abspath(v)
+             for k, v in input_artifact_filepaths.items()}
+        output_artifact_abs_filepaths = \
+            {k: os.path.abspath(v)
+             for k, v in output_artifact_filepaths.items()}
+        job = workflow.to_script(input_artifact_abs_filepaths,
+                                 parameter_references,
+                                 output_artifact_abs_filepaths)
+        with tempfile.NamedTemporaryFile(mode='w', delete=False) as of:
```

Comment:
This is nitpicky, but can you use a more descriptive variable name than `of`? Also, why are you using a tempfile context manager but closing and deleting the file within the context?